src/lightning_fabric/wrappers.py

```diff
@@ -189,3 +190,17 @@ def _process_optimizer_zero_grad_kwargs(optimizer: Optimizer, kwargs: Dict[str,
         # Some optimizers out there, for example DeepSpeedZeroOptimizer, use a different name than PyTorch
         kwargs["set_grads_to_None"] = kwargs.pop("set_to_none")
     return kwargs
+
+
+def _unwrap_objects(collection: T_collection) -> T_collection:
+    def _unwrap(
+        obj: Union[_FabricModule, _FabricOptimizer, _FabricDataLoader]
+    ) -> Union[nn.Module, Optimizer, DataLoader]:
+        if isinstance(obj, _FabricModule):
+            return obj._forward_module
+        if isinstance(obj, _FabricOptimizer):
+            return obj.optimizer
+        if isinstance(obj, _FabricDataLoader):
+            return obj._dataloader
```

Comment:
Should we have a default fallback? or a raise ValueError

```suggestion
            return obj._dataloader
        return obj
```

Reply:
Sure, can't hurt, but not strictly necessary because apply_to_collection won't ever call this function not matching the given types.